pages/index.tsx

```diff
@@ -1,31 +1,123 @@
+import { useState, useEffect, useMemo, useCallback } from 'react';
+import Modal from 'react-modal';
 import styles from '../styles/Home.module.scss';
-import { home_page } from '../config';
+import { home_page, website_config } from '../config';
 // Internal Components
+import Image from '../components/Image';
 import NavBar from '../components/NavBar';
 import EventBanner from '../components/EventBanner';
 import HeroBanner from '../components/HeroBanner';
 import Footer from '../components/Footer';
 import Section from '../layouts/Section';
+import Link from 'next/link';
+
+Modal.setAppElement('#__next'); // Set the app element for accessibility
 
 export default function Home() {
   const { sections } = home_page;
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+
+  useEffect(() => {
+    // Check the referrer when the component mounts
+    const referrer = document.referrer;
+    if (!referrer.includes('tcsca.ca')) {
+      setModalIsOpen(true);
+    }
+  }, []);
+
+  const toggleModal = useCallback(() => {
+    setModalIsOpen((prevState) => !prevState);
+  }, []);
+
+  const renderSections = useMemo(() => {
+    return sections.map((section, i) => (
+      <section key={i}>
+        <Section sectionConfig={section} index={i} />
+      </section>
+    ));
+  }, [sections]);
+
+  const modalContent = useMemo(
```

Comment:
The modal should be its own layout it should really be as simple as just adding `<HackTrentModal>` to the dom because this is a temporary thing it should be super easy to remove.